app/src/main/java/org/fossasia/openevent/app/main/MainActivity.java

```diff
@@ -0,0 +1,169 @@
+        setContentView(R.layout.activity_main);
+        ButterKnife.bind(this);
+
+        navHeader = navigationView.getHeaderView(0);
+        tvEventName = (TextView) navHeader.findViewById(R.id.tvEventName);
+        tvEventTime = (TextView) navHeader.findViewById(R.id.tvEventTime);
+
+        setSupportActionBar(toolbar);
+
+        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
+            this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
+        drawer.addDrawerListener(toggle);
+        toggle.syncState();
+
+        navigationView.setNavigationItemSelectedListener(this);
+        eventId = utilModel.getLong(EVENT_KEY, -1);
+
+        int initialItemId = eventId == -1 ? R.id.nav_events : R.id.nav_dashboard;
+        loadFragment(initialItemId);
+        navigationView.setCheckedItem(initialItemId);
+
+        presenter.attach(this);
+    }
+
+    @Override
+    public void onBackPressed() {
+        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
+        if (drawer.isDrawerOpen(GravityCompat.START)) {
+            drawer.closeDrawer(GravityCompat.START);
+        } else if (backPressed + BACK_PRESS_RESET_TIME > System.currentTimeMillis()) {
+            super.onBackPressed();
+        } else {
+            Toast.makeText(this, "Press again to exit", Toast.LENGTH_SHORT).show();
+        }
+        backPressed = System.currentTimeMillis();
+    }
+
+    @SuppressWarnings("StatementWithEmptyBody")
+    @Override
+    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
+        int id = item.getItemId();
+
+        if (id == R.id.nav_logout) {
+            presenter.logout();
+        } else {
+            if (eventId != -1) {
+                loadFragment(item.getItemId());
+            }
+        }
+
+        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
+        drawer.closeDrawer(GravityCompat.START);
+        return true;
+    }
+
+    @Override
+    public void onLogout() {
+        startActivity(new Intent(this, LoginActivity.class));
+        finish();
+    }
+
+    @Override
+    public void showError(String error) {
+        Toast.makeText(this, error, Toast.LENGTH_SHORT).show();
+    }
+
+    private void loadFragment(int navItemId) {
+        Fragment fragment;
+        switch (navItemId) {
+            case R.id.nav_dashboard:
+                fragment = EventDetailFragment.newInstance(eventId);
+                break;
+            case R.id.nav_attendees:
+                fragment = AttendeesFragment.newInstance(eventId);
+                break;
+            case R.id.nav_events:
+                fragment = EventListFragment.newInstance();
+                break;
+            default:
+                fragment = EventDetailFragment.newInstance(eventId);
+        }
+        setTitle(navigationView.getMenu().findItem(navItemId).getTitle());
+        FragmentManager fm = getSupportFragmentManager();
```

Comment:
Store this in a global variable so that it does not need to get queried again and again